content/crutchcorn/collections/art-of-accessibility/posts/art-of-a11y-text/index.md

```diff
@@ -0,0 +1,561 @@
+
+Can you read it well? How about if you squint? How about from far away?
+
+<p style="opacity: 0.2; filter: blur(2px)">This is hard to read</p>
+
+> This text says the same thing but is now also blurred.
+
+How about _now_?
+
+The reason you're struggling to see this is because it has _low contrast_. For users with [twenty-twenty vision](https://en.wikipedia.org/wiki/Visual_acuity#Definition), the blurred text helps emulate what it _might_ be like to see some text in low contrast for other users.
+
+Let's see what that blurred text looks like with proper color contrast:
+
+ <p style="filter: blur(2px)">This is hard to read</p>
+
+A little easier to see? Good! That's the point. Sufficient contrast can help many users - especially those with color blindness or other visual challenges - see what might otherwise be too difficult to visualize otherwise.
+
+> It's important to note that good contrast doesn't supplement proper screen-reader usage and vice-versa. 
+>
+> Some users may utilize both their eyes and screen reader technology at the same time to help clarify any challenging to see parts of the screen.
+
+## WCAG AA vs AAA
+
+[As mentioned in the preface](/posts/art-of-a11y-preface), there are three levels of compliance with the "Web Content Accessibility Guidelines" (WCAG) - the industry standard guidelines on accessibility: A, AA, and AAA.
+
+While there are no rules in WCAG's A ruleset, AA and AAA both have different requirements for color contrast between a foreground color and background color:
+
+| Ruleset | Content Type | Min Contrast Ratio |
+| --- | --- | --- |
+| AA | Small Text | `4.5:1` |
+| AA | Large Text | `3:1` |
+| AA | Graphical Elements | `3:1` |
+| AAA | Small Text | `7:1` |
+| AAA |Large Text | `4.5:1` |
+| AAA | Graphical Elements| `3:1` |
+
+[This ratio can be calculated either programmatically](https://www.w3.org/TR/WCAG20-TECHS/G17.html#G17-procedure):
+
+```javascript
+const CONTRAST_RATIO = (L1 + 0.05) / (L2 + 0.05)
+```
+
+> L1 is the relative luminance of the lighter of the colors.
+> L2 is the relative luminance of the darker of the colors.
+
+Or manually, using various tools such as [WebAIM's Color Contrast Checker](https://webaim.org/resources/contrastchecker/):
+
+![A contrast checker UI showing the contrast between a blue and a black color](./webaim_checker.png)
+
+## CSS Detection
+
+When comparing AA to AAA compliance, a common qualm lodged against AAA is that its contrast requirements feel too constrictive for a site's "look and feel" compared to AA's easier compliance ratios.
+
+Luckily, there are ways we can have both!
+
+Users are able to opt into an operating system's "high contrast" mode, which will notify the browser to follow in suit.
+
+![Windows 11's "Contrast themes" showing the different styling of high-contrast modes](./windows_11_high_contrast.png)
```

Comment:
These themes apply a [`forced-colors` presentation overriding many authored styles](https://tetralogical.com/blog/2023/04/21/meeting-wcag-level-aaa/#:~:text=There%20is%20no%20separate%20OS%2Dlevel%20option%20on%20Windows%20to%20set%20a%20preference%20for%20higher%20contrast), which makes them not a sufficient technique for passing WCAG's contrast SCs.